prototype_source/fx_numeric_suite_tutorial.py

```diff
@@ -0,0 +1,231 @@
+and its corresponding quantized model to know where are they differ most.
+We built a suite of numeric tools called PyTorch FX Numeric Suite Core APIs in
+PyTorch quantization to enable the measurement of the statistics between
+quantized module and float module to support quantization debugging efforts.
+Even for the quantized model with good accuracy, PyTorch FX Numeric Suite Core
+APIs can still be used as the profiling tool to better understand the
+quantization error within the model and provide the guidance for further
+optimization.
+
+PyTorch FX Numeric Suite Core APIs currently supports models quantized through
+both static quantization and dynamic quantization with unified APIs.
+
+In this tutorial we will use MobileNetV2 as an example to show how to use
+PyTorch FX Numeric Suite Core APIs to measure the statistics between static
+quantized model and float model.
+
+Setup
+^^^^^^
+We’ll start by doing the necessary imports:
+"""
+
+##############################################################################
+
+# Imports and util functions
+
+from __future__ import print_function, division, absolute_import
+import copy
+import torch
+import torchvision
+import torch.quantization
+import torch.ao.ns._numeric_suite_fx as ns
+import torch.quantization.quantize_fx as quantize_fx
+
+import matplotlib.pyplot as plt
+from tabulate import tabulate
+
+torch.manual_seed(0)
+plt.style.use('seaborn-whitegrid')
+
+
+# a simple line graph
+def plot(xdata, ydata, xlabel, ylabel, title):
+    _ = plt.figure(figsize=(10, 5), dpi=100)
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+    plt.title(title)
+    ax = plt.axes()
+    ax.plot(xdata, ydata)
+    plt.show()
+
+##############################################################################
+# Then we load the pretrained float MobileNetV2 model, and quantize it.
+
+
+# create float model
+mobilenetv2_float = torchvision.models.quantization.mobilenet_v2(
+    pretrained=True, quantize=False).eval()
+
+# create quantized model
+qconfig_dict = {
+    '': torch.quantization.get_default_qconfig('fbgemm'),
+    # adjust the qconfig to make the results more interesting to explore
+    'module_name': [
+        # turn off quantization for the first couple of layers
+        ('features.0', None),
+        ('features.1', None),
+        # use MinMaxObserver for `features.17`, this should lead to worse
+        # weight SQNR
+        ('features.17', torch.quantization.default_qconfig),
+    ]
+}
+# Note: quantization APIs are inplace, so we save a copy of the float model for
+# later comparison to the quantized model. This is done throughout the
+# tutorial.
+mobilenetv2_prepared = quantize_fx.prepare_fx(
+    copy.deepcopy(mobilenetv2_float), qconfig_dict)
+datum = torch.randn(1, 3, 224, 224)
+mobilenetv2_prepared(datum)
+# Note: there is a long standing issue that we cannot copy.deepcopy a
+# quantized model. Since quantization APIs are inplace and we need to use
```

Comment:
```suggestion
# quantized model. Since quantization APIs are in place and we need to use
```

Reply:
keeping it as is, lmk if this needs to change